config/nrfconnect/chip-module/Kconfig.features

```diff
@@ -73,6 +73,21 @@ config SPI_NOR_FLASH_LAYOUT_PAGE_SIZE
 
 endif # CHIP_SPI_NOR
 
+config CHIP_MEMORY_PROFILING
+	bool "Enable features for tracking memory usage"
+	default n
+	select CHIP_STATISTICS
+	select CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT
+	select MBEDTLS_MEMORY_DEBUG if !CHIP_CRYPTO_PSA
+	select SYS_HEAP_RUNTIME_STATS
+	select KERNEL_SHELL
+	select NET_STATISTICS
```

Comment:
It's ironic, but enabling these modules to analyze memory usage leads to compilation error for Wi-Fi, due to not enough flash space. I think we have to try reduce the number of enabled modules or resign from enabling profiling on some builds.

Reply:
That's correct. I was able to build the Door Lock with Matter error logs. The light bulb builds fine.